### PR TITLE
[TS] LPS-181899 Item Selector won't show all Sites grouped to the paginated page

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

I have updated the solution as per Brian's [suggestion](https://github.com/brianchandotcom/liferay-portal/pull/134499#issuecomment-1539455549).
The pagination logic is now abstracted out to `ListUtil`.

Best regards,
István

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @IstvanD the [semantic-versioning-jdk8/0/0](https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/624401//consoleText) failure should be related to your PR, could you take a look, thanks!

**Comment by Hanlf** [comment]
ci:test:relevant

File: portal-kernel/src/com/liferay/portal/kernel/util/ListUtil.java [hunk 1]
```diff
@@ -155,6 +156,39 @@ public static <T> List<T> filter(
 		return outputList;
 	}
 
+	public static <T, R> List<T> filter(
+		List<? extends T> inputList, ListFunction<T> listFunction,
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply portal-kernel/src/com/liferay/portal/kernel/util/ListUtil.java:160]: Also, isn't `pageSize` redundant? Its values should always be equal to the initial value of `end - start`.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by IstvanD** [comment]
@adolfopa I pushed the changes.

**Comment by Hanlf** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
The above failures are not related to this PR.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:forward:force

**Pull request closed without merging**